### PR TITLE
Issue#243 shifted spt window

### PR DESCRIPTION
**Pull request by jhmigueles**
The spt window in the first night is shifted to the left in those files starting the recording at midnight. I have tested a couple of changes in g.sib.det and in g.part5 which solve the issue. I have also tested these changes in files starting at different times and the output is not affected.

Best,
Jairo

### CODE REVIEW

**Comment by vincentvanhees** [comment]
Thanks @jhmigueles, I am going to double check it, and if I do not encounter any issues I will merge these changes in the upcoming days.

**Comment by vincentvanhees** [comment]
Hi Jairo,
Issue #243 refers to [the ggir google group](https://groups.google.com/forum/#!msg/rpackageggir/3YBS1HMvUk4/z3ZmvTKRDgAJ) post by Anne Backes, where she shared a milestone data file generated with GGIR part 1 and a plot from part 4. I have now processed that file with your latest code changes, but I am getting the same part4 visualisation as she.
Did you also test your solution with her file? If yes, did the visualisation look different for you?
Best, Vincent

**Comment by jhmigueles** [comment]
Just checked this. The HDCZA algorithm identify correctly the time points for in and out bed estimations with the **sptwindow_HDCZA function**. However, times get wrong for the first day when being passed to "number of hours from last midnight" format (dependent on the initial recording time). To avoid so, I propose to directly look for the bed in and out times identified by the HDZCA algorithm in the "time" vector and then transform this to "number of hours" format (just in this first night when the initial recording time is different to midnight).

Let me know if anything is unclear.

Best,
Jairo

File: R/g.sib.det.R [hunk 1]
```diff
@@ -279,8 +279,21 @@ g.sib.det = function(M,IMP,I,twd=c(-12,12),anglethreshold = 5,
                                      perc = perc, inbedthreshold = inbedthreshold, 
                                      bedblocksize = bedblocksize, outofbedsize = outofbedsize)
           if (length(inbedout$sptwindow_HDCZA_end) != 0 & length(inbedout$sptwindow_HDCZA_start) != 0) {
-            sptwindow_HDCZA_end[j] = (inbedout$sptwindow_HDCZA_end / (3600/ ws3)) + 12
-            sptwindow_HDCZA_start[j] = (inbedout$sptwindow_HDCZA_start / (3600/ ws3)) + 12
+            if(j == 1 & midnightsi[j] == 1) {
+              sptwindow_HDCZA_end[j] = (inbedout$sptwindow_HDCZA_end/(3600/ws3))
+              sptwindow_HDCZA_start[j] = (inbedout$sptwindow_HDCZA_start/(3600/ws3))
+            } else if(j == 1){   #added 02/12/2019. First night get shifted start and end bedtimes depending on the initial time
```
[the remark this answers is not in the document]
Comment (jhmigueles) [reply R/g.sib.det.R]: Thanks Vincent!

Just done all your requested changes.

Best,
Jairo

[the remark this answers is not in the document]
Comment (vincentvanhees) [reply R/g.sib.det.R]: thanks Jairo for making these changes so quickly, did you forget this one?

File: R/g.sib.det.R [hunk 2]
```diff
@@ -279,8 +279,21 @@ g.sib.det = function(M,IMP,I,twd=c(-12,12),anglethreshold = 5,
...
+              sptwindow_HDCZA_start[j] = (inbedout$sptwindow_HDCZA_start/(3600/ws3))
+            } else if(j == 1){   #added 02/12/2019. First night get shifted start and end bedtimes depending on the initial time 
+              end_t = unlist(iso8601chartime2POSIX(time[inbedout$sptwindow_HDCZA_end], tz = desiredtz))
+              end = as.numeric(end_t["hour"]) + as.numeric(end_t["min"])/60 + as.numeric(end_t["sec"])/60/60
+              sptwindow_HDCZA_end[j] = end
+
+              start_t = unlist(iso8601chartime2POSIX(time[inbedout$sptwindow_HDCZA_start], tz = desiredtz))
+              start = as.numeric(start_t["hour"]) + as.numeric(start_t["min"])/60 + as.numeric(start_t["sec"])/60/60
+              sptwindow_HDCZA_start[j] = start
+            } else {
```
[the remark this answers is not in the document]
Comment (vincentvanhees) [reply R/g.sib.det.R]: missed this one too? if you have already done it then ignore this. Later today I will merge all of it in the master.

**Comment by vincentvanhees** [comment]
Thanks Jairo, All looks great. When the tests on Travis pass I will merge this in the Master and prepare new release.

File: R/g.sib.det.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhmigueles) [reply R/g.sib.det.R]: I was working in two separated tabs and some changes just dissapeared. I resolved it just after sending my previous message, it should be now ok (same for the othe change that you couldn't see).

**Pull request merged**